DataFormats/Portable/interface/PortableCollectionCommon.h

```diff
@@ -0,0 +1,113 @@
+
+#ifndef DataFormats_Portable_interface_PortableCollectionCommon_h
+#define DataFormats_Portable_interface_PortableCollectionCommon_h
+
+template <auto Start, auto End, auto Inc, class F>
+constexpr void constexpr_for(F&& f) {
+  if constexpr (Start < End) {
+    f(std::integral_constant<decltype(Start), Start>());
+    constexpr_for<Start + Inc, End, Inc>(f);
```

Comment:
I'm just wondering if these should be
```suggestion
    std::forward<F>(f)(std::integral_constant<decltype(Start), Start>());
    constexpr_for<Start + Inc, End, Inc>(std::move(f));
```
to support rvalue `f` ?

Reply:
Also, I would use `std::size_t` explicitly instead of `auto`, and give a default value of `Inc = 1`